### PR TITLE
Remove quoted.Type synthesization

### PR DESCRIPTION
**Pull request by nicolasstucki**
* Add `scala.quoted.Type.apply[T]`. Just as `$t` is equivalent to `t.T`, `'[T]` is equivalent to `Type[T]`.
* Remove the implicit search synthesizer for `quoted.Type`
* Simplify printing

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala [hunk 1]
```diff
@@ -266,14 +266,13 @@ class RefinedPrinter(_ctx: Context) extends PlainPrinter(_ctx) {
     ("{" ~ toText(trees, "\n") ~ "}").close
 
   protected def typeApplyText[T >: Untyped](tree: TypeApply[T]): Text = {
-    val isQuote = !printDebug && tree.fun.hasType && tree.fun.symbol == defn.InternalQuoted_typeQuote
-    val (open, close) = if (isQuote) (keywordStr("'["), keywordStr("]")) else ("[", "]")
-    val funText = toTextLocal(tree.fun).provided(!isQuote)
+    val isQuote = !printDebug && tree.fun.hasType && tree.fun.symbol == defn.QuotedTypeModule_apply
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala:269]: `isQuote` is not used.

File: compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala [hunk 2]
```diff
@@ -175,8 +175,10 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
       tp match
         case tp: TypeRef =>
           tp.prefix match
-            case NoPrefix if level > levelOf(tp.symbol) =>
-              tryHeal(tp.symbol, tp, pos)
+            case NoPrefix if level > levelOf(tp.symbol) && !tp.typeSymbol.hasAnnotation(defn.InternalQuoted_QuoteTypeTagAnnot) =>
+              val tp1 = tp.dealias
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala:179]: What about making `tp.dealias` as the scrutinee of the top-level pattern match?

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala]: Done

File: compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala (hunk 1 above)
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala]: Removed

File: compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala [hunk 3]
```diff
@@ -175,8 +175,10 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
       tp match
         case tp: TypeRef =>
           tp.prefix match
-            case NoPrefix if level > levelOf(tp.symbol) =>
-              tryHeal(tp.symbol, tp, pos)
+            case NoPrefix if level > levelOf(tp.symbol) && !tp.typeSymbol.hasAnnotation(defn.InternalQuoted_QuoteTypeTagAnnot) =>
+              tp.dealias match
+                case `tp` => tryHeal(tp.symbol, tp, pos)
+                case tp1 => apply(tp1)
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala:181]: I wasn't clear in the previous comment. I mean what about doing dealiasing at line 175?

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala:181]: No, that is too much dealiasing. There are a few aliases that we care about, such as the type splices.
I will revert this change.

**Pull request merged**